src/broker/groups.c

```diff
@@ -123,7 +123,7 @@ static struct group *group_lookup (struct groups *g,
             return NULL;
         if (zhashx_insert (g->groups, group->name, group) < 0) {
             group_destroy (group);
-            errno = ENOMEM;
+            errno = EEXIST;
```

Comment:
This is inside a block that has already checked whether the group name is in the hash, so I think the insert return code can be ignored.